testing/asserts.ts

```diff
@@ -864,3 +866,58 @@ export function unimplemented(msg?: string): never {
 export function unreachable(): never {
   throw new AssertionError("unreachable");
 }
+
+let snapshotFile: Record<string, string> | undefined = undefined;
+const updatedSnapshotFile: Record<string, unknown> = {};
+const snapshotMap: Record<string, number> = {};
+
+export async function assertSnapshot(context: Deno.TestContext, actual: unknown) {
+  const name = getName(context);
+  const count = getCount();
+  const testName = `${name} #${count}`;
+  const isUpdate = Deno.args.includes('--update');
+  if (!snapshotFile) {
+    const snapshotPath = getSnapshotPath();
+    await ensureFile(snapshotPath);
+    const {snapshot} = await import(snapshotPath);
+    snapshotFile = snapshot;
+  }
+  const snapshot = snapshotFile?.[testName] ?? '';
+  if (isUpdate) {
+    let isEqual = true;
+    try {
+      assertEquals(_format(actual), snapshot.slice(1, -1));
+    } catch {
+      isEqual = false;
+    }
+    if (!isEqual) console.info("Snapshot updated", testName);
+    updatedSnapshotFile[testName] = actual;
+    globalThis.onunload = writeSnapshotFileSync;
```

Comment:
Seems like this function will be run once per `assertSnapshot` call. Can we refactor this so it only runs once?

Reply:
Done